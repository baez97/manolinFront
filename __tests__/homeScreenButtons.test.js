// import { SCLAlert, SCLAlertButton } from 'react-native-scl-alert';
import React from 'react';
import renderer from 'react-test-renderer';
import GlobalButton from '../components/homeComponents/globalButton';
import MidBlueButton from '../components/homeComponents/midBlueButton';
import MidYellowButton  from '../components/homeComponents/midYellowButton';


jest.unmock("react-native")
jest.unmock('react')
jest.unmock("react-test-renderer")
jest.unmock('expo');

it('GlobalButton renders correctly', () => {
    const rendered = renderer.create(<GlobalButton />).toJSON();
    expect(rendered).toMatchSnapshot();
});

it('MidBlueButton renders correctly', () => {
    const rendered = renderer.create(<MidBlueButton />).toJSON();
    expect(rendered).toMatchSnapshot();
});

it('GlobalButton renders correctly', () => {
    const rendered = renderer.create(<MidYellowButton />).toJSON();
    expect(rendered).toMatchSnapshot();
});



