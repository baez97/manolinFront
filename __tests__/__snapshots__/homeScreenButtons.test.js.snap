// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalButton renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <ExponentLinearGradient
    colors={
      Array [
        4279286482,
        4280914167,
      ]
    }
    endPoint={
      Array [
        0.65,
        1,
      ]
    }
    startPoint={
      Array [
        0.55,
        0,
      ]
    }
    style={
      Object {
        "borderRadius": 20,
        "elevation": 15,
        "flexDirection": "row",
        "height": 120,
        "justifyContent": "space-evenly",
        "marginTop": 20,
        "padding": 10,
        "width": 400,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontFamily": "montserrat-extra-bold",
          "fontSize": 21,
          "marginTop": 10,
          "textAlign": "left",
        }
      }
    />
    <Image
      source={1}
      style={
        Object {
          "height": 100,
          "width": 100,
        }
      }
    />
  </ExponentLinearGradient>
</View>
`;

exports[`GlobalButton renders correctly 2`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <ExponentLinearGradient
    colors={
      Array [
        4294954035,
        4294089034,
      ]
    }
    endPoint={
      Array [
        0.65,
        1,
      ]
    }
    startPoint={
      Array [
        0.55,
        0,
      ]
    }
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 20,
        "elevation": 15,
        "height": 80,
        "justifyContent": "center",
        "marginTop": 20,
        "width": 190,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#3f2606",
          "fontFamily": "montserrat-extra-bold",
          "fontSize": 20,
          "textAlign": "left",
        }
      }
    />
  </ExponentLinearGradient>
</View>
`;

exports[`MidBlueButton renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <ExponentLinearGradient
    colors={
      Array [
        4279286482,
        4278481116,
      ]
    }
    endPoint={
      Array [
        0.65,
        1,
      ]
    }
    startPoint={
      Array [
        0.55,
        0,
      ]
    }
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 20,
        "elevation": 15,
        "height": 80,
        "justifyContent": "center",
        "marginTop": 20,
        "width": 190,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontFamily": "montserrat-extra-bold",
          "fontSize": 20,
          "textAlign": "left",
        }
      }
    />
  </ExponentLinearGradient>
</View>
`;
